InMobiAdapter/src/main/java/com/chartboost/mediation/inmobiadapter/InMobiAdapter.kt

```diff
@@ -384,6 +368,44 @@ class InMobiAdapter : PartnerAdapter {
         }
     }
 
+    /**
+     * Build a [JSONObject] that will be passed to the InMobi SDK for GDPR during [setUp].
+     *
+     * @param gdprConsent A Boolean indicating whether GDPR consent is granted or not.
+     * @param context The current [Context].
+     *
+     * @return a [JSONObject] object as to whether GDPR consent is granted or not.
+     */
+    private fun buildGdprJsonObject(gdprConsent: Boolean, context: Context): JSONObject {
+        return JSONObject().apply {
+            try {
+                put(InMobiSdk.IM_GDPR_CONSENT_AVAILABLE, gdprConsent)
+                put("gdpr", if (gdprApplies == true) "1" else "0")
+                getTcfString(context).takeIf { it.isNotEmpty() }?.let { tcfString ->
+                    put(InMobiSdk.IM_GDPR_CONSENT_IAB, tcfString)
+                } ?: PartnerLogController.log(CUSTOM, "TCFv2 String is empty or was not found.")
```

Comment:
Nit: I don't quite like the use of `takeIf { }` + `let { }` here because we're going from something that's non-null (the result of `getTcfString()`) to something that's nullable (the result of `takeIf { }`) unnecessarily. And then we have to `let { }` that. It's a bit of an overkill. 

Here's one of those situations where a classic if-else might make it more readable:

```kt
val tcfString = getTCFString(context)
if (tcfString.isNotEmpty()) {
    put(GDPR_CONSENT_IAB_KEY, tcfString)
} else {
    PartnerLogController.log(CUSTOM, "TCFv2 String is empty or was not found.")
}
```

Reply:
Sure. We can have it  a bit more readable. 👍🏼